automation/utils.ts

```diff
@@ -23,6 +23,69 @@ import * as shellEscape from 'shell-escape';
 export const MSYS2_BASH = 'C:\\msys64\\usr\\bin\\bash.exe';
 export const ROOT = path.join(__dirname, '..');
 
+/** Tap and buffer this process' stdout and stderr */
+export class StdOutTap {
+	public stdoutBuf: string[] = [];
+	public stderrBuf: string[] = [];
+	public allBuf: string[] = [];
+
+	protected origStdoutWrite: typeof process.stdout.write;
+	protected origStderrWrite: typeof process.stdout.write;
+
+	constructor(protected printDots = false) {}
```

Comment:
This class captures `stdout` and `stderr` for comparison against `pkg`'s saved output. If `printDots` is `true`, then the output actually printed to the console is replaced with dots, like in the long line of dots below:
```
$ npm run build
...
Building standalone zip package for CLI v12.12.1
...
...............................................................................
```
The output capture is the real output rather than the dots, of course.
I thought that it would be nicer to print dots than the `pkg` warnings that make people think that something is wrong:
```
> Warning Cannot include file %1 into executable.
  The file must be distributed with executable as %2.
  %1: node_modules/drivelist/build/Release/drivelist.node
  %2: path-to-executable/drivelist.node
```
This is OK now that the known safe output is saved to a file and automatically checked.

Reply:
Replacing the known errors with dots is very nice.